### DISCUSSION TITLE
Question regarding applying cnt_thresh

### DISCUSSION THREAD

**Discussion by jwolff-ncar** [dtcenter/METplus#976] [Configuration] [answered]
We have added logic into our METplus conf to look at REFC values >15 (dBZ)

Forecast/Observation variable Information
FCST_VAR1_NAME = REFC
FCST_VAR1_LEVELS = L0
**FCST_VAR1_OPTIONS = cnt_thresh = [ >15 ];**

BOTH_VAR1_THRESH = ge20, ge30, ge40, ge50

OBS_VAR1_NAME = MergedReflectivityQComposite
OBS_VAR1_LEVELS = L0
**OBS_VAR1_OPTIONS = censor_thresh = eq-999; censor_val = -9999; cnt_thresh = [ >15 ]; cnt_logic = UNION;**

But when we look at FBAR values in the CNT line we see times when FBAR is <15. Why would this be if we have set the cnt_thresh >15? This does not happen in all cases, but wanted to raise the question to see if we may be missing something in the process.

V10.0.0 FV3_RRFS_v1alpha_3km_summer NA 010000 20190530_010000 20190530_010000 000000 20190530_010041 20190530_010041 REFC dB L0 MergedReflectivityQComposite dBZ L0 MRMS CONUS NEAREST 1 **>15|| >15** NA 0.05 CNT 102781 **11.6506** 11.5394 11.7618 NA NA **18.18924** 18.11095 18.26822 NA NA 0.53792 0.23347 0.84237 NA NA 49.79922 49.58488 50.01544 NA NA -0.32409 -0.32955 -0.3186 NA NA NA NA 0 0 0 11.11268 10.75631 11.46905 NA NA 58.29178 58.04088 58.54487 NA NA 21.65863 NA NA 41.85589 NA NA 3521.3904 NA NA 3397.89868 NA NA 59.3413 NA NA -40.7854 NA NA -28.13012 NA NA -9.85971 NA NA 15.1155 NA NA 119.95 NA NA 43.24562 NA NA 20.54083 NA NA NA NA NA NA NA 123.49172 NA NA -0.41994 NA NA NA NA NA NA NA NA NA NA NA

**Comment by jprestop** [comment]
@jwolff-ncar Hi Jamie.  Thanks for submitting this here.  Can you please tell us what MET tool(s) you are using in your METplus runs?  I believe that @JohnHalleyGotway or @georgemccabe would best able to help you.  They are both out today, but I will post in the Slack user-support forum and have someone follow up with you as soon as possible.

**Comment by jwolff-ncar** [comment]
@jprestop @JohnHalleyGotway @georgemccabe For this particular task we are just running GridStat. 

This was run on hera. The output file is here: /scratch2/BMC/fv3lam/RRFS_baseline/expt_dirs/RRFS_baseline_summer/2019053000/metprd/grid_stat/grid_stat_FV3_RRFS_v1alpha_3km_summer_REFC_MRMS_010000L_20190530_010000V.stat

The conf file is here:
/scratch2/BMC/fv3lam/RRFS_baseline/ufs-srweather-app/regional_workflow/ush/templates/parm/metplus/REFC.conf

**Comment by JohnHalleyGotway** [comment, marked as answer]
Jamie, I see you have a question about how the cnt_thresh option is applied in Grid-Stat. You have defined cnt_thresh = [ >15 ]; to compute continuous statistics, but only for reflectivity values (REFC) greater than 15. Look closely at the options you've defined below:
```
FCST_VAR1_OPTIONS = cnt_thresh = [ >15 ];
OBS_VAR1_OPTIONS = censor_thresh = eq-999; censor_val = -9999; cnt_thresh = [ >15 ]; cnt_logic = UNION;
```
This tells 
Grid-Stat to include all matched pairs where the forecast value is >15 **OR** the observation value is >15. That's what the "cnt_logic = UNION;" is telling it to do. If your forecast has a low-bias, it may very well be the case that the forecast values are <15 while the observation values are >15... and that would explain why you're seeing FBAR values less than 15.

It's up to you to decide exactly what filtering logic you'd like to use. Some other options you might consider are listed below. In the examples listed below, note that the NA threshold always evaluates to true.

(1) Forecast and observation values both >15:
```
FCST_VAR1_OPTIONS = cnt_thresh = [ >15 ];
OBS_VAR1_OPTIONS = cnt_thresh = [ >15 ]; cnt_logic = INTERSECTION;
```
(2) Condition only on the forecast values being >15:
```
FCST_VAR1_OPTIONS = cnt_thresh = [ >15 ];
OBS_VAR1_OPTIONS = cnt_thresh = [ NA ]; cnt_logic = INTERSECTION;
```
(3) Condition only on the observation values being >15:
```
FCST_VAR1_OPTIONS = cnt_thresh = [ NA ];
OBS_VAR1_OPTIONS = cnt_thresh = [ >15 ]; cnt_logic = INTERSECTION;
```

Note that specifying the NA threshold isn't actually necessary. When the cnt_thresh arrays for the forecast and observation fields have uneven length, they are automatically padded with NA thresholds until the lengths match. I just listed them here to be explicit about it.

You could also do all 3 options listed above in a single run, like this:
```
FCST_VAR1_OPTIONS = cnt_thresh = [ >15, >15, NA ];
OBS_VAR1_OPTIONS = cnt_thresh = [ >15, NA, >15 ]; cnt_logic = INTERSECTION;
```

If you want to employ different cnt_logic options, you can still do so in a single run... BUT you'd need to do it using different VAR options. By that I mean have VAR1 use cnt_logic = INTERSECTION and VAR2 use cnt_logic = UNION.

If this answers your question, please consider clicking the checkmark at the top of this post to indicate that.

**Comment by jwolff-ncar** [comment]
@JohnHalleyGotway Thank you for the helpful response. It makes complete sense.
